File: Attendance/Models/Attendance.cs
using Attendance.Models.SharedProp;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;


namespace Attendance.Models
{
    public class Attendance : Audit
    {
        public int Id { get; set; }
        public DateTime CheckIn { get; set; }
        public DateTime CheckOut { get; set; }
        public string Status { get; set; }
        public TimeSpan WorkingHours 
        { 
            get
            {
                return CheckOut - CheckIn;
            }
        }

        public int UserId { get; set; }
        public User User { get; set; }
    }
}

File: Attendance/Angular/src/app/shared/models/attendance.ts
export class Attendance {
  Id?: number;
  UserId?: number;
  CheckIn?: string | null; // Allow null here
  CheckOut?: string | null; // Allow null here
  Date?: string| null; // Allow null here
  WorkingHours?: number;
  Status?: string;
  TotalHours? : Number;
}
